### DISCUSSION TITLE
Advice on Adding a Permissions Layer

### DISCUSSION THREAD

**Discussion by jonaird** [geldata/gel#2045] [Q&A] [answered]
Hi, 
As I understand it permissions are on the roadmap. I’m trying to figure out the best way to add a permissions layer between a front end and the built in GraphQL api. My first guess would be to add a Nodejs server in between and use Apollo. Any thought?

Thanks!

**Comment by elprans** [comment, marked as answer]
Policy-based access control is high on the list of priorities.  In fact, we've already laid most of the initial groundwork for it in 7f29af4c, and the remaining bits are mostly about adding the necessary schema syntax and support for session variables.

Before that lands, the best way I can think of is to make a middleware that rewrites incoming GraphQL queries.  Running an Apollo proxy might work.
